### PR TITLE
moved HighlightOverlay to VisualizationBase

### PR DESCRIPTION
**Pull request by mgalbier**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58073652%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58073817%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58073959%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58074020%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23issuecomment-203987088%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58170643%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23issuecomment-204298342%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23issuecomment-203987088%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22One%20more%20thing%20you%20need%20to%20do%3A%20Each%20visualization%20comes%20with%20at%20least%20one%20stlye.%20Please%20move%20all%20styles%20that%20are%20defined%20by%20default%20to%20the%20new%20location.%20Styles%20for%20an%20item%20are%20in%20%60PluginName/styles/item/ItemTypeName/%2A%60%22%2C%20%22created_at%22%3A%20%222016-03-31T15%3A29%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Alright%2C%20this%20looks%20good.%22%2C%20%22created_at%22%3A%20%222016-04-01T07%3A53%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2081bc3cf875290c37866514a3aa1b636f1cd29e0e%20VisualizationBase/CMakeLists.txt%2035%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58073652%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20changed%20on%20this%20line%3F%20I%20don%27t%20see%20a%20difference.%22%2C%20%22created_at%22%3A%20%222016-03-31T15%3A25%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20have%20no%20idea.%20I%27m%20pretty%20sure%20I%20didn%27t%20touch%20that%20line%22%2C%20%22created_at%22%3A%20%222016-04-01T07%3A27%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VisualizationBase/CMakeLists.txt%3AL268-272%22%7D%2C%20%22Pull%2081bc3cf875290c37866514a3aa1b636f1cd29e0e%20VisualizationBase/src/overlays/HighlightOverlayStyle.h%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58074020%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22relative%20paths%22%2C%20%22created_at%22%3A%20%222016-03-31T15%3A27%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VisualizationBase/src/overlays/HighlightOverlayStyle.h%3AL26-40%22%7D%2C%20%22Pull%2081bc3cf875290c37866514a3aa1b636f1cd29e0e%20VisualizationBase/src/overlays/HighlightOverlay.cpp%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58073817%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22All%20includes%20from%20the%20same%20plugin%2C%20should%20use%20relative%20paths.%5Cr%5Cn%5Cr%5CnPlease%20look%20a%20few%20lines%20above%20this%20one%20and%20make%20the%20paths%20relative%20to%20the%20current%20file.%22%2C%20%22created_at%22%3A%20%222016-03-31T15%3A26%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VisualizationBase/src/overlays/HighlightOverlay.cpp%3AL34-41%22%7D%2C%20%22Pull%2081bc3cf875290c37866514a3aa1b636f1cd29e0e%20VisualizationBase/src/overlays/HighlightOverlay.h%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/359%23discussion_r58073959%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22As%20for%20the%20other%20files%3A%20relative%20paths%20in%20includes.%22%2C%20%22created_at%22%3A%20%222016-03-31T15%3A27%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VisualizationBase/src/overlays/HighlightOverlay.h%3AL26-33%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 81bc3cf875290c37866514a3aa1b636f1cd29e0e VisualizationBase/CMakeLists.txt 35'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/359#discussion_r58073652'>File: VisualizationBase/CMakeLists.txt:L268-272</a></b>
- <a href='https://github.com/dimitar-asenov'></a> What changed on this line? I don't see a difference.
- <a href='https://github.com/mgalbier'></a> I have no idea. I'm pretty sure I didn't touch that line
- [ ] <a href='#crh-comment-Pull 81bc3cf875290c37866514a3aa1b636f1cd29e0e VisualizationBase/src/overlays/HighlightOverlay.cpp 5'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/359#discussion_r58073817'>File: VisualizationBase/src/overlays/HighlightOverlay.cpp:L34-41</a></b>
- <a href='https://github.com/dimitar-asenov'></a> All includes from the same plugin, should use relative paths.
  Please look a few lines above this one and make the paths relative to the current file.
- [ ] <a href='#crh-comment-Pull 81bc3cf875290c37866514a3aa1b636f1cd29e0e VisualizationBase/src/overlays/HighlightOverlay.h 5'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/359#discussion_r58073959'>File: VisualizationBase/src/overlays/HighlightOverlay.h:L26-33</a></b>
- <a href='https://github.com/dimitar-asenov'></a> As for the other files: relative paths in includes.
- [ ] <a href='#crh-comment-Pull 81bc3cf875290c37866514a3aa1b636f1cd29e0e VisualizationBase/src/overlays/HighlightOverlayStyle.h 5'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/359#discussion_r58074020'>File: VisualizationBase/src/overlays/HighlightOverlayStyle.h:L26-40</a></b>
- <a href='https://github.com/dimitar-asenov'></a> relative paths
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/359#issuecomment-203987088'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> One more thing you need to do: Each visualization comes with at least one stlye. Please move all styles that are defined by default to the new location. Styles for an item are in `PluginName/styles/item/ItemTypeName/*`
- <a href='https://github.com/dimitar-asenov'></a> Alright, this looks good.

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/359?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/359?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/359'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/359'>Refresh</a>.<a href='#crh-update-none'></a>

File: VisualizationBase/CMakeLists.txt [hunk 1]
```diff
@@ -264,4 +268,4 @@ endif (WIN32)
 find_package(Qt5 REQUIRED Xml Svg PrintSupport WebKitWidgets)
 target_link_libraries(VisualizationBase -llpsolve55 Qt5::Xml Qt5::Svg Qt5::PrintSupport Qt5::WebKitWidgets)
 
-envision_plugin(VisualizationBase ModelBase)
\ No newline at end of file
+envision_plugin(VisualizationBase ModelBase)
```
Comment (dimitar-asenov) [inline VisualizationBase/CMakeLists.txt:271]: What changed on this line? I don't see a difference.

File: VisualizationBase/src/overlays/HighlightOverlay.cpp [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
 using namespace Visualization;
 
-namespace InformationScripting {
+namespace Visualization {
```
Comment (dimitar-asenov) [inline VisualizationBase/src/overlays/HighlightOverlay.cpp:36]: All includes from the same plugin, should use relative paths.

Please look a few lines above this one and make the paths relative to the current file.

File: VisualizationBase/src/overlays/HighlightOverlay.h [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 #pragma once
 
-#include "../informationscripting_api.h"
+#include "../visualizationbase_api.h"
```
Comment (dimitar-asenov) [inline VisualizationBase/src/overlays/HighlightOverlay.h:29]: As for the other files: relative paths in includes.

File: VisualizationBase/src/overlays/HighlightOverlayStyle.h [hunk 4]
```diff
@@ -26,14 +26,14 @@
 
 #pragma once
 
-#include "../informationscripting_api.h"
+#include "../visualizationbase_api.h"
```
Comment (dimitar-asenov) [inline VisualizationBase/src/overlays/HighlightOverlayStyle.h:29]: relative paths

**Comment by dimitar-asenov** [comment]
One more thing you need to do: Each visualization comes with at least one stlye. Please move all styles that are defined by default to the new location. Styles for an item are in `PluginName/styles/item/ItemTypeName/*`

File: VisualizationBase/CMakeLists.txt (hunk 1 above)
Comment (mgalbier) [inline VisualizationBase/CMakeLists.txt:271]: I have no idea. I'm pretty sure I didn't touch that line.

**Comment by dimitar-asenov** [comment]
Alright, this looks good.

**Pull request merged**